src/components/TextField/TextArea.js

```diff
@@ -0,0 +1,279 @@
+    wrapInputWithTooltip = (textarea, tooltip) => {
+        const { DEFAULT, ERROR } = TooltipBoxStatus;
+        return (
+            tooltip ?
+                <Tooltip
+                    position={TooltipPosition.BOTTOM_CENTER}
+                    showTooltip={this.state.showTooltip}
+                    tooltip={
+                        <TooltipBox
+                            status={this.props.tooltipError ? ERROR : DEFAULT}
+                        >
+                            {tooltip}
+                        </TooltipBox>
+                    }
+                >
+                    {textarea}
+                </Tooltip> :
+                textarea
+        );
+    }
+
+    render() {
+        /* eslint-disable jsx-a11y/label-has-for */
+        // @NB: jsx-a11y/label-has-for fails with UID as id
+        const showLabel = (
+            this.props.hasLabelAlways ||
+            this.state.hasFocus ||
+            this.state.hasMouseOver ||
+            !this.state.value
+        );
+        const requiredIcon = this.props.isRequired ?
+            <Tooltip tooltip={this.props.tooltipRequired}><IconRequired /></Tooltip> :
+            null;
+        const label = this.props.label && showLabel ?
+            (
+                <label htmlFor={this.uid} className="uir-textarea-label">
+                    {this.props.label}
+                    {requiredIcon}
+                </label>
+            ) :
+            null;
+        const showPlaceholder = (!this.props.label || this.state.hasFocus);
+        return (
+            <div
+                className={cx(
+                    'uir-textarea',
+                    {
+                        'uir-textarea--disabled': this.props.isDisabled,
+                        'uir-textarea--focus': this.state.hasFocus,
+                        'uir-textarea--full-width': this.props.isFullWidth,
+                        'uir-textarea--has-auto-height': this.props.hasAutoHeight,
+                        'uir-textarea--has-right-icon': this.props.isRequired,
+                        'uir-textarea--has-value': this.state.value,
+                        'uir-textarea--invalid': this.props.isValid === false,
+                        'uir-textarea--readonly': this.props.isReadOnly,
+                        'uir-textarea--valid': this.props.isValid,
+                    },
+                    this.props.className,
+                )}
+                onMouseEnter={this.handleMouseEnter}
+                onMouseLeave={this.handleMouseLeave}
+                style={this.props.style}
+            >
+                <div className="uir-textarea-inner">
+                    <div className="uir-textarea-label-wrapper">
+                        {label}
+                    </div>
+                    {this.wrapInputWithTooltip(
+                        <textarea
+                            aria-invalid={this.props.isValid === false}
+                            aria-required={this.props.isRequired}
```

Comment:
MDN says this:

> Now that required is available to more than 97% of users globally, it is no longer recommended that you use both required and aria-required. 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/Basic_form_hints#Required_and_invalid_fields

Reply:
good call, have updated